test/IntegrationTest.php

```diff
@@ -75,6 +81,99 @@ public function testResolveIPv6only()
+                    $config->getNameservers(),
+                    $config->getKnownHosts(),
+                    $config->getTimeout(),
+                    $config->getAttempts(),
+                    ['kelunik.com'],
+                    1
+                )));
+
+            Dns\resolver(new Dns\Rfc1035StubResolver(null, $configLoader));
+            $result = yield Dns\resolve('blog');
+
+            /** @var Record $record */
+            $record = $result[0];
+            $inAddr = @\inet_pton($record->getValue());
+            $this->assertNotFalse(
+                $inAddr,
+                "Server name blog.kelunik.com did not resolve to a valid IP address"
+            );
+        });
+    }
+
+    public function testFailResolveRootedDomainWhenSearchListDefined()
+    {
+        Loop::run(function () {
+            $configLoader = \stripos(PHP_OS, "win") === 0
+                ? new WindowsConfigLoader()
+                : new UnixConfigLoader();
+            /** @var Dns\Config $config */
+            $config = yield $configLoader->loadConfig();
+            /** @var Dns\ConfigLoader|MockObject $configLoader */
+            $configLoader = $this->createMock(Dns\ConfigLoader::class);
+            $configLoader->expects($this->once())
+                ->method('loadConfig')
+                ->willReturn(new Success(new Dns\Config(
+                    $config->getNameservers(),
+                    $config->getKnownHosts(),
+                    $config->getTimeout(),
+                    $config->getAttempts(),
+                    ['kelunik.com'],
+                    1
+                )));
+
+            Dns\resolver(new Dns\Rfc1035StubResolver(null, $configLoader));
+            $this->expectException(DnsException::class);
+            yield Dns\resolve('blog.');
+        });
+    }
+
+    public function testResolveWithRotateList()
+    {
+        Loop::run(function () {
+            /** @var Dns\ConfigLoader|MockObject $configLoader */
+            $configLoader = $this->createMock(Dns\ConfigLoader::class);
+            $configLoader->expects($this->once())
+                ->method('loadConfig')
+                ->willReturn(new Success(new Dns\Config(
+                    [
+                        '208.67.222.220:53', // Opendns
+                        '195.243.214.4:53', // Deutche Telecom AG
+                    ],
+                    [],
+                    5000,
+                    3,
+                    [],
+                    1,
+                    true
+                )));
+
+            $resolver = new Dns\Rfc1035StubResolver(new NullCache(), $configLoader);
+
+            /** @var Record $record1 */
+            list($record1) = yield $resolver->query('facebook.com', Dns\Record::A);
+            /** @var Record $record2 */
+            list($record2) = yield $resolver->query('facebook.com', Dns\Record::A);
+
+
+            $this->assertNotSame($record1->getValue(), $record2->getValue());
```

Comment:
This doesn't look like a very stable test, they might resolve to the same or a different value at any time, no?

Reply:
@kelunik These are nameservers from different continents, they mostly resolve different IP records due to geolocalisation. I see no other option to test this feature or should I remove the test?

Reply:
@brzuchal Let's keep it for now, we can always change it if it beaks.